### PR TITLE
1165-incompatible-browser-validation-and-warning

### PR DESCRIPTION
**Pull request by Dmac26**
﻿## Summary
Addresses Issue # 1165
This PR is an updated version of the original 1165 PR, which needed several styling updates to account for mobile views and to clean up some syntax. I was unable to directly update the original PR as I did not manage my work flow properly and my branch was significantly behind the dev branch.

## This pull request is ready to merge when...
- [x] Feature branch is starts with the issue number
- [x] Is connected to its original issue via zenhub 👇
- [x] All tests are passing and meet coverage, linting, and accessibility requirements. And no security vulnerabilities ⚫️(Jenkins)
- [ ] This code has been reviewed by someone other than the original author

### CODE REVIEW

**Comment by Dmac26** [comment]
Closing PR to attempt another PR post move from Circle to Jenkins.

**Comment by Dmac26** [comment]
@briandavidson - Hey Brian, just a heads up. I sent you a message detailing multiple questions and problems I'm having with the current PRs en queue. Gratsi sir!

File: frontend/src/app/app.component.ts [hunk 1]
```diff
@@ -131,5 +91,28 @@ export class AppComponent implements OnInit {
...
+        if (browserInfo[1] === 'Chrome') {
+        parsedBrowserInfo = userAgent.match (/\bOPR|Edge\/(\d+)/);
+        if (parsedBrowserInfo != null)   {
+            return { name: 'Opera', version: parsedBrowserInfo[1] };
+          }
+        }
+      browserInfo = browserInfo[2] ? [browserInfo[1], browserInfo[2]] : [navigator.appName, navigator.appVersion, '-?'];
+      if (( parsedBrowserInfo = userAgent.match(/version\/(\d+)/i)) != null) {
+        browserInfo.splice(1, 1, parsedBrowserInfo[1]);
+    }
```
Comment (briandavidson) [inline frontend/src/app/app.component.ts:114]: Please fix the code style here to be consistent with the rest of the codebase. The indentation is off on a few lines.

File: frontend/src/sass/components/_header.scss [hunk 2]
```diff
@@ -144,3 +144,15 @@
   margin-top: 0.5rem;
   margin-bottom: 0.5rem;
 }
+
+.browser-validation-main-content {
+  @media (min-width: 319px) {
+    margin-top: 18rem;
+  }
+  @media (min-width:365px) {
```
Comment (briandavidson) [inline frontend/src/sass/components/_header.scss:152]: Missing a space after `min-width:`

Comment (Dmac26) [reply frontend/src/sass/components/_header.scss]: Added missing space back in.

**Comment by Dmac26** [comment]
@briandavidson @mtlaney - Hey Mike and Brian, this PR should be good for a final review.

File: frontend/src/app/app.component.ts [hunk 3]
```diff
@@ -131,5 +91,28 @@ export class AppComponent implements OnInit {
         return hour < 12 ? 'AM' : 'PM';
       }
     });
-   }
+  }
+
+  getBrowserName() {
+    const  userAgent = navigator.userAgent;
```
Comment (briandavidson) [inline frontend/src/app/app.component.ts:97]: looks like an extra space after `const`

File: frontend/src/app/app.component.ts [hunk 4]
```diff
@@ -131,5 +91,28 @@ export class AppComponent implements OnInit {
...
+      return { name: 'IE', version: (parsedBrowserInfo[1] || '') };
+    }
+    if (browserInfo[1] === 'Chrome') {
+      parsedBrowserInfo = userAgent.match (/\bOPR|Edge\/(\d+)/);
+      if (parsedBrowserInfo != null)   {
+        return { name: 'Opera', version: parsedBrowserInfo[1] };
+      }
+    }
+    browserInfo = browserInfo[2] ? [browserInfo[1], browserInfo[2]] : [navigator.appName, navigator.appVersion, '-?'];
+    if (( parsedBrowserInfo = userAgent.match(/version\/(\d+)/i)) != null) {
```
Comment (briandavidson) [inline frontend/src/app/app.component.ts:112]: This line works just fine, but for code style consistency can you please remove the blank spaces between the `(` and the inner contents?

To be honest I'd prefer the same treatment applied to the `{` brackets as well, but I think you might find both ways of writing it in the codebase.

File: frontend/src/app/app.component.ts [hunk 5]
```diff
@@ -131,5 +91,28 @@ export class AppComponent implements OnInit {
...
+      parsedBrowserInfo = /\brv[ :]+(\d+)/g.exec(userAgent) || [];
+      return { name: 'IE', version: (parsedBrowserInfo[1] || '') };
+    }
+    if (browserInfo[1] === 'Chrome') {
+      parsedBrowserInfo = userAgent.match (/\bOPR|Edge\/(\d+)/);
+      if (parsedBrowserInfo != null)   {
+        return { name: 'Opera', version: parsedBrowserInfo[1] };
+      }
+    }
+    browserInfo = browserInfo[2] ? [browserInfo[1], browserInfo[2]] : [navigator.appName, navigator.appVersion, '-?'];
```
Comment (briandavidson) [inline frontend/src/app/app.component.ts:111]: @Dmac26  What is going on here, does this work? Are you intentionally setting `browserInfo` equal to `browserInfo[2]` before your ternary or should it be a `===` to compare the values?

Comment (Dmac26) [reply frontend/src/app/app.component.ts]: @briandavidson - While looking at this to determine what I had in mind with that code, I remembered that the majority of this code snippet was borrowed from another developer on Shadat's team. I didn't quite understand everything that was going on with it, so I took a bit of a deeper dive and then decided to rewrite most of it to suite our needs. I believe that what is going on in my new method is far more apparent and too the point. Let me know what you think!

File: frontend/src/app/app.component.ts [hunk 6]
```diff
@@ -131,5 +91,29 @@ export class AppComponent implements OnInit {
...
+          }
+          case 'Edge': {
+            this.browserName = 'Edge';
+            break;
+          }
+          default: {
+            this.browserName = 'notChrome';
+            break;
+          }
+        }
```
Comment (briandavidson) [inline frontend/src/app/app.component.ts:115]: This `switch` is much cleaner, nice work here @Dmac26!

**Pull request merged**